multi-agent/scripts/migrate_duplicate_blueprint_names.py

```
@@ -0,0 +1,116 @@
+              f"duplicates={len(duplicates)}")
+
+        # Collect all existing names for this identity so we don't collide
+        # with names that already exist outside this duplicate group.
+        existing_names = set(
+            doc["spec_dict"]["name"]
+            for doc in col.find(
+                {"identity.type": g["_id"]["identity_type"],
+                 "identity.id": identity_id},
+                {"spec_dict.name": 1},
+            )
+        )
```

Ruff (0.15.21)

[warning] Unnecessary generator (rewrite as a set comprehension)

Rewrite as a set comprehension

(C401)